test_e2e.sh

```diff
@@ -1658,7 +1658,7 @@ function e2e::sync_depth_across_updates() {
     assert_metric_eq "${METRIC_GOOD_SYNC_COUNT}" 1
     assert_metric_eq "${METRIC_FETCH_COUNT}" 1
     depth=$(git -C "$ROOT/link" rev-list HEAD | wc -l)
-    if [[ $expected_depth != $depth ]]; then
+    if [[ $expected_depth != "$depth" ]]; then
```

Comment:
We should probably quote both sides

Reply:
`shellcheck` was probably not detecting this because `expected_depth` is getting set only on two occasions to the static value 1:

```
rg expected_depth=
test_e2e.sh
1644:    expected_depth="1"
2709:    expected_depth="1"
```

Anyways, I agree with quoting it to keep the code consistent. Fixed!